hasktorch/src/Torch/Functional.hs

```diff
@@ -943,3 +943,26 @@ flattenAll
   -> Tensor -- ^ output
 flattenAll t =
   unsafePerformIO $ (cast3 ATen.flatten_tll) t (0 :: Int) (-1 :: Int)
+
+-- | smoothL1Loss
+smoothL1Loss
+  :: Int -- ^ reduction
+  -> Tensor -- ^ target
+  -> Tensor -- ^ input
+  -> Tensor -- ^ output
+smoothL1Loss reduction target input = unsafePerformIO $ (cast3 ATen.smooth_l1_loss_ttl) input target reduction
+
+-- | softMarginLoss
+softMarginLoss 
+  :: Int -- ^ reduction
+  -> Tensor -- ^ target
+  -> Tensor -- ^ self
+  -> Tensor -- ^ output
+softMarginLoss reduction target input = unsafePerformIO $ (cast3 ATen.soft_margin_loss_ttl) input target reduction
```

Comment:
We need to create a data type that encodes the reductions, because it's completely unclear what's reduction 0 and what's reduction 1 (plus, there's no reduction 289346). You'll need to add something along the lines of:
```hs
data Reduction = Option1 | Option2 | ...

instance Castable Reduction Int where
  ...
```

Reply:
Alright.

Reply:
@apaszke Please have a look now? The datatype for reduction was already present. Had to just use it in the functions.